src/app/modules/item/task/rxjschannel.ts

```diff
@@ -0,0 +1,75 @@
+import { build, ChannelConfiguration, MessageTransaction, MessagingChannel } from 'jschannel';
+import { Observable } from 'rxjs';
+
+export interface RxMessage<T> {
+  method: string;
+  params?: T;
+  timeout?: number;
+  error?: (error: any, message: string) => void;
+}
+
+/** Build a RxMessagingChannel, which is a jschannel with rxjs calls */
+export function rxBuild(config: ChannelConfiguration): Observable<RxMessagingChannel> {
+  return new Observable<RxMessagingChannel>(subscriber => {
+    let chan : RxMessagingChannel | null = null;
+    const innerConfig = {
+      onReady: () : void => {
+        subscriber.next(chan as RxMessagingChannel);
+        subscriber.complete();
+      },
+      ...config
+    };
+    chan = new RxMessagingChannel(build(innerConfig));
+  });
+}
+
+export class RxMessagingChannel {
+  innerChan: MessagingChannel;
+
+  constructor(innerChan: MessagingChannel) {
+    this.innerChan = innerChan;
+  }
+
+  unbind(method: string, doNotPublish?: boolean): boolean {
+    return this.innerChan.unbind(method, doNotPublish);
+  }
+
+  bind<T>(method: string, observable?: (params: T) => Observable<any>, doNotPublish?: boolean): MessagingChannel {
+    // Create a callback wrapping the observable bound
+    function callback(transaction: MessageTransaction, params: T): void {
+      if (!observable) {
+        return;
+      }
+      const cb$ = observable(params);
+      cb$.subscribe({
+        next: transaction.complete,
```

Comment:
`next` is not `complete`, it may be called several times.
```suggestion
        complete: transaction.complete,
```

Reply:
`transaction.complete` from jschannel needs to receive the return value from the Observable, and is supposed to be called only once. Should I add a take(1) to ensure next is called only once?

Reply:
> Should I add a take(1) to ensure next is called only once?

Yes

Reply:
Done.